go-controller/pkg/ovn/egressip.go

```diff
@@ -2503,15 +2597,15 @@ func (e *EgressIPController) getNextHop(ni util.NetInfo, egressNodeName, egressI
 	isEgressIPv6 := utilnet.IsIPv6String(egressIP)
 	// NOTE: No need to check if status.node exists or not in the cache, we are calling this function only if it
 	// is present in the nodeZoneState cache. Since we call it with lock on cache, we are safe here.
-	if isLocalZoneEgressNode {
+	if isLocalZoneEgressNode || ni.TopologyType() == types.Layer2Topology {
 		if isOVNNetwork {
```

Comment:
can you add to doc text on this function what isOVNNetwork means? Everytime I see this variable i forget what it is

Reply:
I think it means is the egress IP served by an OVS bridge (sgw mode) vs is it on interface not part of an OVS/OVN topology